packages/block-library/src/navigation-link/link-ui.js

```diff
@@ -186,54 +200,145 @@ export function LinkUI( props ) {
-			onClose={ props.onClose }
-			anchor={ props.anchor }
-			shift
-		>
-			<LinkControl
-				hasTextControl
-				hasRichPreviews
-				value={ link }
-				showInitialSuggestions={ true }
-				withCreateSuggestion={ userCanCreate }
-				createSuggestion={ handleCreate }
-				createSuggestionButtonText={ ( searchTerm ) => {
-					let format;
-
-					if ( type === 'post' ) {
-						/* translators: %s: search term. */
-						format = __( 'Create draft post: <mark>%s</mark>' );
-					} else {
-						/* translators: %s: search term. */
-						format = __( 'Create draft page: <mark>%s</mark>' );
-					}
-
-					return createInterpolateElement(
-						sprintf( format, searchTerm ),
-						{
-							mark: <mark />,
-						}
-					);
-				} }
-				noDirectEntry={ !! type }
-				noURLSuggestion={ !! type }
-				suggestionsQuery={ getSuggestionsQuery( type, kind ) }
-				onChange={ props.onChange }
-				onRemove={ props.onRemove }
-				onCancel={ props.onCancel }
-				renderControlBottom={
-					! url
-						? () => (
-								<LinkControlTransforms
-									clientId={ props.clientId }
-								/>
-						  )
-						: null
-				}
-			/>
-		</Popover>
+		<>
+			{ showBackdrop && (
+				<div
+					className="components-popover-pointer-events-trap"
+					aria-hidden="true"
+					onClick={ () => setShowBackdrop( false ) }
+				/>
+			) }
+			<Popover
+				placement="bottom"
+				onClose={ props.onClose }
+				anchor={ props.anchor }
+				shift
+			>
+				{ ! addingBlock && (
+					<div
+						role="dialog"
+						aria-labelledby={ dialogTitleId }
+						aria-describedby={ dialogDescritionId }
+					>
+						<VisuallyHidden>
+							<h2 id={ dialogTitleId }>{ __( 'Add link' ) }</h2>
+
+							<p id={ dialogDescritionId }>
+								{ __(
+									'Search for and add a link to your Navigation.'
+								) }
+							</p>
+						</VisuallyHidden>
+						<LinkControl
+							hasTextControl
+							hasRichPreviews
+							value={ link }
+							showInitialSuggestions={ true }
+							withCreateSuggestion={ userCanCreate }
+							createSuggestion={ handleCreate }
+							createSuggestionButtonText={ ( searchTerm ) => {
+								let format;
+
+								if ( type === 'post' ) {
+									/* translators: %s: search term. */
+									format = __(
+										'Create draft post: <mark>%s</mark>'
+									);
+								} else {
+									/* translators: %s: search term. */
+									format = __(
+										'Create draft page: <mark>%s</mark>'
+									);
+								}
+
+								return createInterpolateElement(
+									sprintf( format, searchTerm ),
+									{
+										mark: <mark />,
+									}
+								);
+							} }
+							noDirectEntry={ !! type }
+							noURLSuggestion={ !! type }
+							suggestionsQuery={ getSuggestionsQuery(
+								type,
+								kind
+							) }
+							onChange={ props.onChange }
+							onRemove={ props.onRemove }
+							onCancel={ props.onCancel }
+							renderControlBottom={ () =>
+								! link?.url?.length && (
```

Comment:
We need to conditionally render controls only when we're not creating a page.

Otherwise on slow connections you can end up in strange situations when you click "Create page" and then are still able to click "Add block" within the Link UI.


https://github.com/WordPress/gutenberg/assets/444434/8af859e0-7929-4fac-9ca4-ad1abc6cb137

Reply:
Pushed a fix for this in a2d0a972a068c46e3bc7d36bd384053e9737e445